distributed/nanny.py

```diff
@@ -259,9 +256,17 @@ async def start(self):
 
         await super().start()
 
+        self._start_address = addresses_from_user_args(
+            host=self._start_host,
+            port=parse_worker_ports(self._start_port),
```

Comment:
I'm not sure that this is what we want.  If I provide a range like "8000:9000" then I think that this will provide a list of 1000 addresses on which to listen.  Is that correct?  If so, that could quickly become problematic. 

Instead, I would expect the port logic to occur around when we try to listen on a port, and find it to be busy.

Reply:
It will provide a list of 1000 possible addresses to listen on but when we then make the call to `self.listen` it will start to loop through the 1000 addresses until we find one that is open, create a single listener for that address, and then return. So each `listen` call results in, at most, one listener being created. There's an added test, `distributed/tests/test_core.py::test_multiple_addresses`, which demonstrates this behavior. 

I agree if we created 1000 listeners here that would be problematic. Do you think the behavior outlined above is still problematic?

Reply:
OK, I think that there might be a slight conflict here that may be my fault.  

In https://github.com/dask/distributed/pull/3288 I added the ability for the Scheduler to listen on many addresses at once.  This is where the `addresses_from_user_args` function came from.  However I never took the time to extend this functionality over to worker/nanny.  The intention here was that we would iterate over many addresses from `self._start_address` and listen on all of them.

In this PR we're now reusing that same mechanism, but for a different purpose saying "listen on the first one that works".  Everything today is fine, but the moment that we try to either have multiple networks active on the workers, or use a port range on the scheduler, we'll run into a biggish conflict I think.

Reply:
Alright, thanks for the additional context. We would indeed have a conflict if we wanted to use, say, a port range and multiple interfaces on a `Worker`. 

I'll roll back the recent changes that moved the "listen on the first address that works" logic to `Server.listen` and instead keep an explicit `for`-loop over multiple ports in the `Worker`/`Nanny` classes like in https://github.com/dask/distributed/pull/3704/commits/2d7162e305bfc531c23f07391ba2c7e14efaaa4e. I think having a single input address for `Server.listen` will help us better isolate any port range and/or multiple interface logic moving forward.